### ISSUE TITLE
Loss curves and intermediate visualizations

### ISSUE THREAD

**Issue by Jiawei-Yang** [facebookresearch/capi#19]
Hi,

Thanks for the great work. I'm currently training CAPI at a smaller scale, and I'm unsure if these loss curves look normal to you.

![Image](https://github.com/user-attachments/assets/c8fc10f1-48a0-459e-b86a-e2949396b6a2)

Specifically, I'm training CAPI-ViT-Large on the ImageNet-22K dataset. The learning rate schedule has been adjusted accordingly for 100k iterations (I simply linearly scale down the iteration-based schedulers). These curves don't look quite like the one in Figure 9 (capi paper). Is this expected, or should I train the model for longer?

When I visualized feature maps from these models, I got results like the following:
![Image](https://github.com/user-attachments/assets/45cbc1bf-919e-4b94-a307-e0b5a875ecc4)
From left to right: original image, feature PCA visualized as RGB, the first 6 PCA components, and feature norm (visualized at iteration 14,999).

Below are more examples with similarly fluctuating loss curves:

![Image](https://github.com/user-attachments/assets/eba5f005-2226-4801-b6be-08a95d2789a3)

Thanks!
jiawei

**Comment by Jiawei-Yang** [comment]
For reference, my schedules are:
optim.total_iters=100000
optim.lr_schedule.warmup_iters=10000
optim.momentum_schedule.warmup_iters=10000
capi.clustering_optimizer.lr_schedule.warmup_iters=10000
for 100k iters training, and

optim.total_iters=5000
optim.lr_schedule.warmup_iters=5000
optim.momentum_schedule.warmup_iters=5000
capi.clustering_optimizer.lr_schedule.warmup_iters=5000
for 50k iters training

**Comment by TimDarcet** [comment]
Hi,

No, these curves look very unhealthy. The warmup may be too fast.

To get a sanity check, I'd advise first launching a run with all the normal hparams and schedules, including the 50k warmup (even if you don't let it run to the end):
- If you don't have issues, then you just need to keep a longer warmup (maybe 20k-30k could be enough).
- If you do see issues, it means something deeper is at play (maybe the dataset? How did you load IN22k, is it the HF implem in the codebase?).

**Comment by Jiawei-Yang** [comment]
hey Tim,

Thanks for the advice! I will look into this issue soon once i'm available. 

For the dataset, i'm using the IN22k HF implementation in the codebase. As far as i can remember, the only change I made to the dataset loading part is adding a lambda function to convert images to "RGB" format for the HF implementation.

Thanks!
best,
jiawei

**Comment by Jiawei-Yang** [comment]
Hi,

I'm unsure what I had changed. I guess it's something related to torch.device("meta") etc. But the loss curves look much more reasonable to me now. Thank you!

I'm closing this issue now.

**Comment by TimDarcet** [comment]
Great to hear! Maybe the weight init did not take place and so the initial weights were wild random values
